src/Microsoft.DotNet.Build.Tasks/AddDependenciesToProjectJson.cs

```diff
@@ -83,41 +126,51 @@ private static JObject ReadProject(string projectJsonPath)
            } 
         }
 
+        private JToken GetFrameworkDependenciesSection(JObject projectJsonRoot, string framework = null)
+        {
+            if(string.IsNullOrWhiteSpace(framework))
```

Comment:
To your first point, correct.  It's more important to me that I understand the scenario you were describing for the default (empty) case, but this is good cleanup.

To your second point, is this an issue?  Is this also just cleanup?  I haven't noticed any problem with duplicate entries in the generated json.  

{
  "dependencies": {
    "Microsoft.NETCore.Platforms": "1.0.1-rc3-24011-00",
    "System.Globalization": "4.0.11-rc3-24011-00",
    "System.Globalization.Calendars": "4.0.1-rc3-24011-00",
    "System.Linq.Expressions": "4.0.11-rc3-24011-00",
    "System.ObjectModel": "4.0.12-rc3-24011-00",
    "System.Runtime.Extensions": "4.1.0-rc3-24011-00",
    "System.Text.RegularExpressions": "4.0.12-rc3-24011-00",
    "System.Threading.Tasks": "4.0.11-rc3-24011-00",
    "xunit": "2.1.0",
    "xunit.netcore.extensions": "1.0.0-prerelease-00231-07"
  },
  "frameworks": {
    "dnxcore50": {
      "imports": "portable-net45+win8",
      "dependencies": {
        "System.Globalization.Calendars": "4.0.1-rc3-24011-00"
      }
    },
    "net46": {
      "dependencies": {
        "Microsoft.TargetingPack.NETFramework.v4.6": "1.0.1",
        "System.Globalization.Calendars": "4.0.1-rc3-24011-00"
      }
    }
  }
}